crates/ntx-builder/src/actor.rs

```diff
@@ -0,0 +1,248 @@
+use std::sync::Arc;
+
+use futures::FutureExt;
+use miden_node_proto::domain::account::NetworkAccountPrefix;
+use miden_node_proto::domain::mempool::MempoolEvent;
+use miden_node_utils::ErrorReport;
+use miden_objects::account::Account;
+use miden_objects::transaction::TransactionId;
+use miden_remote_prover_client::remote_prover::tx_prover::RemoteTransactionProver;
+use tokio::sync::{AcquireError, RwLock, Semaphore, mpsc};
+use tokio_util::sync::CancellationToken;
+use url::Url;
+
+use crate::block_producer::BlockProducerClient;
+use crate::builder::ChainState;
+use crate::state::{State, TransactionCandidate};
+use crate::store::StoreClient;
+use crate::transaction::NtxError;
+
+/// The reason an actor has shut down.
+pub enum ActorShutdownReason {
+    AccountReverted(NetworkAccountPrefix),
+    EventChannelClosed,
+    SemaphoreFailed(AcquireError),
+    Cancelled(NetworkAccountPrefix),
+}
```

Comment:
Could we add doc comments to enum variants to explain what they mean? For some of them the meaning is simple to infer, but, for example, for `SemaphoreFailed` and `Cancelled` it is not clear why these would be triggered.